public/app/features/plugins/admin/state/selectors.ts

```diff
@@ -32,11 +33,14 @@ export type PluginFilters = {
 export const selectPlugins = (filters: PluginFilters) =>
   createSelector(selectAll, (plugins) => {
     const keyword = filters.keyword ? unEscapeStringFromRegex(filters.keyword.toLowerCase()) : '';
+    const filteredPluginIds = keyword !== '' ? filterByKeyword(plugins, keyword) : null;
 
     return plugins.filter((plugin) => {
-      const fieldsToSearchIn = [plugin.name, plugin.orgName].filter(Boolean).map((f) => f.toLowerCase());
+      if (keyword && filteredPluginIds == null) {
+        return false;
+      }
 
-      if (keyword && !fieldsToSearchIn.some((f) => f.includes(keyword))) {
+      if (keyword && filteredPluginIds != null && !filteredPluginIds.includes(plugin.id)) {
```

Comment:
nit: here we already know `filteredPluginIds` is not null because is checked in the previous condition, so you can remove the check here.